dist/gaia-header.js

```diff
@@ -913,11 +918,40 @@ module.exports = component.register('gaia-header', {
   observerStop: function() {
     if (!this.observing) { return; }
     this.observer.disconnect();
+    window.removeEventListener('resize', this);
+    this._resizeHandled = false;
```

Comment:
I tried to think of the scenarios I was protecting from, and I couldn't really find a compelling scenario. So I'll remove these state resets. Usually I try to reset state wherever I want but here it seems unnecessary (and even maybe incorrect).